src/hotspot/share/runtime/continuation.cpp

```diff
@@ -2196,30 +2195,31 @@ inline intptr_t* Thaw<ConfigT>::thaw(thaw_kind kind) {
   assert(chunk != nullptr, "guaranteed by prepare_thaw");
   assert(!chunk->is_empty(), "guaranteed by prepare_thaw");
 
+  // I have no idea what config does in this function.
   _barriers = chunk->requires_barriers();
   return (LIKELY(can_thaw_fast(chunk))) ? thaw_fast(chunk)
                                         : thaw_slow(chunk, kind != thaw_top);
 }
 
 template <typename ConfigT>
 NOINLINE intptr_t* Thaw<ConfigT>::thaw_fast(stackChunkOop chunk) {
-  assert(chunk != nullptr, "");
   assert(chunk == _cont.tail(), "");
-  assert(!chunk->is_empty(), "");
   assert(!chunk->has_mixed_frames(), "");
   assert(!chunk->requires_barriers(), "");
   assert(!chunk->has_bitmap(), "");
   assert(!_thread->is_interp_only_mode(), "");
 
   // TODO: explain why we're not setting the tail
 
+  // LogTarget might generate some test code
   LogTarget(Trace, continuations) lt;
   if (lt.develop_is_enabled()) {
     LogStream ls(lt);
     ls.print_cr("thaw_fast");
     chunk->print_on(true, &ls);
   }
 
+  // Comment needed: why is 500? Related to 300 in can_thaw_fast?
```

Comment:
It's a heuristic. Below this number, we thaw the whole chunk. Above it, we thaw one frame. 500 seemed to work ok, but might need to be adjusted after deeper profiling.

Reply:
It's unrelated. The 300 is just an estimate for the stack size taken by the native code. 500 is just a heuristic of when to thaw one frame or the entire chunk (so refers to Java frames). The function always makes room for the entire chunk.

Reply:
ok.  I'll try to think of something to say.